src/components/burger-menu/BurgerMenu.jsx

```diff
@@ -1,6 +1,74 @@
-import React, { useState, useContext } from 'react';
+import React, { useContext, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { GlobalContext } from '../GlobalContext';
+import { AppRoutes, HeaderItems } from '../../config';
+import styles from './BurgerMenu.module.scss';
+import burgerMenuIcon from '../../menu-icon/menu-icon.svg';
+import closeMenuIcon from '../../menu-icon/close-menu-icon.svg';
+import createClassNames from '../../createClassNames';
 
 export const BurgerMenu = () => {
-  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [isOpen, setIsOpen] = useState(false);
+  const { amounts: { accumulations }, balanceState } = useContext(GlobalContext);
 
-}
\ No newline at end of file
+  const navigate = useNavigate();
+  const location = useLocation();
+  const pathName = location.pathname === '/' ? AppRoutes.MY_WALLET : location.pathname;
+
+  const menuItemHandler = (route) => {
+    navigate(route);
+    setIsOpen(false);
+  };
+
+  const menuOpenHandler = () => {
+    setIsOpen(!isOpen);
```

Comment:
setIsOpen(prevValue => !prevValue);
becasue when you'll click really fast 'isOpen' won't have valid value, coz react does rerender asynchronously, 
if you want to get valid previous value use callback in setState hook.